src/librustc/middle/cstore.rs

```diff
@@ -474,7 +477,13 @@ pub mod tls {
     {
         let tls_payload = (ecx as *const _, encoder as *mut _);
         let tls_ptr = &tls_payload as *const _ as *const TlsPayload;
-        TLS_ENCODING.set(unsafe { &*tls_ptr }, || f(ecx, encoder))
+        TLS_ENCODING.with(|tls| {
+            let prev = tls.get();
```

Comment:
Hm. I don't personally care that much (this is pretty simple stuff) but I remember @nikomatsakis saying he wanted to continue using scoped TLS. He's probably not going to be around before this needs to land, though. I think we can probably take it as-is and figure out a better story later if needed.